### PR TITLE
Add collector cmd example

### PR DESCRIPTION
**Pull request by lumjjb**
Add the collector command example to show how a collector binary would be able to run a collector, currently emit is not implemented due to undecided mechanism between collector and processor.

### CODE REVIEW

**Review by mihaimaruseac — APPROVED** [review]

**Review by trmiller — APPROVED** [review]
LGTM

File: cmd/collector/cmd/example.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+		ctx := context.Background()
+
+		// Register collector
+		// We create our own MockCollector and register it
+		mc := mockcollector.NewMockCollector()
+		collector.RegisterDocumentCollector(mc, mc.Type())
+
+		// Collect
+		docChan, errChan, numCollectors, err := collector.Collect(ctx)
```
[the remark this answers is not in the document]
Comment (lumjjb) [reply cmd/collector/cmd/example.go:45]: yea that's fair, thoughts on moving this entire thing into `Collect` and hide all this channel stuff?

Like make it `Collect(ctx context.Context, emitter processor.Emitter)`

```
type Emitter func(*processor.Document) error
```

[the remark this answers is not in the document]
Comment (lumjjb) [reply cmd/collector/cmd/example.go:45]: will leave this to another PR.

[the remark this answers is not in the document]
Comment (pxp928) [reply cmd/collector/cmd/example.go:45]: Yes I like that approach better. Good to merge this for now and create an issue to add the changes you mentioned above.

[the remark this answers is not in the document]
Comment (lumjjb) [reply cmd/collector/cmd/example.go:45]: opened https://github.com/guacsec/guac/issues/32

**Pull request merged**